test/functional/test_framework/test_framework.py

```diff
@@ -532,12 +532,15 @@ def skip_if_no_wallet(self):
             raise SkipTest("wallet has not been compiled.")
 
     def skip_if_no_cli(self):
-        """Skip the running test if bitcoin-cli has not been compiled."""
-        if not self.is_cli_compiled():
+        """Skip the running test if bitcoin-cli is not available."""
+        if not self.is_cli_available():
             raise SkipTest("bitcoin-cli has not been compiled.")
 
-    def is_cli_compiled(self):
-        """Checks whether bitcoin-cli was compiled."""
+    def is_cli_available(self):
+        """Checks whether bitcoin-cli is available."""
+        if os.getenv("BITCOINCLI", "__NOT_SET__") != "__NOT_SET__":
```

Comment:
Why not just - `if os.getenv("BITCOINCLI") != None:`?

Reply:
In both Python 2 and 3 docs, default is `None` as well.
* https://docs.python.org/2/library/os.html#os.getenv 
* https://docs.python.org/3/library/os.html#os.getenv

Reply:
"key, default and the result are str."

Reply:
Default being `None` is documented.

Reply:
Not as a return value.

Reply:
I agree with @kristapsk here. The correct way to do this is:
```python
if os.getenv("BITCOINCLI") is not None:
```
or, if you don't want to support an empty value / regard an empty value as unset
```python
if not os.getenv("BITCOINCLI"):
```
Special-casing is a default value is, imo, fragile and ugly.

Reply:
Special-casing the default here is well-defined, not fragile... the alternatives are what is fragile, since they're not guaranteed behaviours [yet?].

Reply:
Returning `None` or a string is just the pythonic way to say it returns `optional<string>`

Reply:
> Returning None or a string is just the pythonic way to say it returns optional<string>

Exactly.